src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs

```diff
@@ -241,12 +235,28 @@ public void Close()
     /// <inheritdoc/>
     public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryCompleteEventArgs, IEnumerable<TestCase> lastChunk)
     {
-        _baseTestDiscoveryEventsHandler.HandleDiscoveryComplete(discoveryCompleteEventArgs, lastChunk);
+        if (lastChunk != null)
+        {
+            // When discovery is complete then the last discovered source is still marked
+            // as partially discovered, so we need to mark it as fully discovered.
+            _discoverySourceStatusCache.MarkTheLastChunkSourcesAsFullyDiscovered(lastChunk);
+        }
```

Comment:
I don't like this nullcheck, if there were no extra sources to be discovered and we did not abort, then it would be logical to send null. Is this just a substitute for checking IsAborted?